components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -5,3 +5,597 @@
+	end
+
+	return record
+end
+
+function CustomMatchGroupInput._readTeamPlayers(opponent, playerData)
+	local players = CustomMatchGroupInput._getManuallyEnteredPlayers(playerData)
+	opponent.players = nil
+
+	if Table.isEmpty(players) then
+		players = CustomMatchGroupInput._getPlayersFromVariables(opponent.name)
+	end
+
+	return players
+end
+
+function CustomMatchGroupInput._getManuallyEnteredPlayers(playerData)
+	local players = {}
+	playerData = Json.parseIfString(playerData) or {}
+
+	for prefix, displayName in Table.iter.pairsByPrefix(playerData, 'p') do
+		local name = mw.ext.TeamLiquidIntegration.resolve_redirect(Logic.emptyOr(
+			playerData[prefix .. 'link'],
+			displayName
+		))
+
+		table.insert(players, {
+			name = name,
+			displayname = displayName,
+			flag = Flags.CountryName(playerData[prefix .. 'flag']),
+		})
+	end
+
+	return players
+end
+
+function CustomMatchGroupInput._getPlayersFromVariables(teamName)
+	local players = {}
+
+	local playerIndex = 1
+	local prefix = teamName .. '_p' .. playerIndex
+	local playerName = Variables.varDefault(prefix)
+	while String.isNotEmpty(playerName) do
+		table.insert(players, {
+			name = playerName,
+			displayname = Variables.varDefault(prefix .. 'dn'),
+			flag = Flags.CountryName(Variables.varDefault(prefix .. 'flag')),
+		})
+
+		playerIndex = playerIndex + 1
+		prefix = teamName .. '_p' .. playerIndex
+		playerName = Variables.varDefault(prefix)
+	end
+
+	return players
+end
+
+--[[
+
+MapInput functions
+
+]]--
+function CustomMatchGroupInput._mapInput(match, mapIndex, subGroupIndex)
+	local map = Json.parseIfString(match['map' .. mapIndex]) or {}
+
+	if Table.isEmpty(map) then
+		match['map' .. mapIndex] = nil
+		return match, subGroupIndex
+	end
+
+	-- CR has no map names, use generic one instead
+	map.map = 'Set ' .. mapIndex
+
+	-- set initial extradata for maps
+	map.extradata = {
+		comment = map.comment,
+		header = map.header,
+	}
+
+	-- inherit stuff from match data
+	map.type = Logic.emptyOr(map.type, match.type)
+	map.liquipediatier = match.liquipediatier
+	map.liquipediatiertype = match.liquipediatiertype
+	map.game = Logic.emptyOr(map.game, match.game)
+	map.date = Logic.emptyOr(map.date, match.date)
+
+	-- determine score, resulttype, walkover and winner
+	map = CustomMatchGroupInput._mapWinnerProcessing(map)
+
+	-- get participants data for the map + get map mode
+	map = CustomMatchGroupInput._processPlayerMapData(map, match)
+
+	-- set sumscore to 0 if it isn't a number
+	if String.isEmpty(match.opponent1.sumscore) then
+		match.opponent1.sumscore = 0
+	end
+	if String.isEmpty(match.opponent2.sumscore) then
+		match.opponent2.sumscore = 0
+	end
+
+	--adjust sumscore for winner opponent
+	if (tonumber(map.winner) or 0) > 0 then
+		match['opponent' .. map.winner].sumscore =
+			match['opponent' .. map.winner].sumscore + 1
+	end
+
+	match['map' .. mapIndex] = map
+
+	return match, subGroupIndex
+end
+
+function CustomMatchGroupInput._mapWinnerProcessing(map)
```

Comment:
`_matchWinnerProcessing` and `_mapWinnerProcessing` should have the same logic (barring bestOf) for setting status, score, finished, winner etc

Reply:
why?
matches have actual scores
maps like on sc/sc2 maps if played have a winner (0, 1, 2) but no scores